src/vm/functions/database.rs

```diff
@@ -74,6 +74,10 @@ pub fn special_fetch_contract_entry(args: &[SymbolicExpression],
 pub fn special_set_entry(args: &[SymbolicExpression],
                          env: &mut Environment,
                          context: &LocalContext) -> Result<Value> {
+    if env.global_context.is_read_only() {
```

Comment:
Due to the type checker, this should never be true, right?  Just want to make sure I'm interpreting this error branch as an instance of defensive coding, as opposed to an error branch that a developer may need to contend with.

Reply:
Yes, that's right, this should never happen in code that made it past the type checker.

Reply:
We should definitely further classify interpreter errors (in the rust code itself) as ones that might be expected during runtime and ones that should not be expected during runtime.